### DISCUSSION TITLE
Is there a way to change forced username format in Coder OSS

### DISCUSSION THREAD

**Discussion by azat-alimov-db** [coder/coder#13340] [General] [answered]
Hello,

We are trying to integrate Coder OSS with Github EMU and the usernames format there contains "_"(underscore). 
Coder is not allowing to have usernames with such symbol and hence throwing an error: "Username must start with a-Z or 0-9 and can contain a-Z, 0-9 or -" for new users, that are trying to sign-up, using Github OAuth app. 

Appreciate if someone can help or point out to guidance on how to either change that rule or use some other field as a username or crop the existing emu handle to remove anything after "_"?

**Comment by kylecarbs** [comment]
Hmm, I believe we allow _ in the middle, just like GitHub.

If our validation does not match GitHubs, we'll fix it.

Can you provide me with a sample username that is valid on GitHub, but not on Coder?

**Reply by azat-alimov-db** [reply]
Yes, sure. For example my github handle in EMU is azat-alimov_dbk. Before integrating with github I had an internal Coder username as azatalimov. Post Github integration for authentication I was able to login successfully, since Coder didn't intend to change my username, to one that I have in github. But when new users are trying to use Github as authentication provider, Coder tries to sync their username (firs-last_dbk) and fails with the username format error.
Coder version is 2.11.1

**Reply by azat-alimov-db** [reply]
@kylecarbs 
Hey Kyle, any news on that?

**Reply by kylecarbs** [reply]
I see it's only allowed for Enterprise users... kinda jank for us to allow this.

Is it blocking you?

**Reply by azat-alimov-db** [reply]
Not a blocker, really, I still can create users manually or via cli with allowed username format, but with github type authentication.

Would be nice to have and ability to change the forced username format in Coder. Just a potential enhancement request for future.

Thanks

**Comment by matifali** [comment, marked as answer]
Hi @azat-alimov-db, this was resolved in #13384  and is available as per `v2.13.x` and later versions. Please test and see if you see this.

**Reply by azat-alimov-db** [reply]
That's great, thank you I will test it and will let you know

**Reply by matifali** [reply]
It is not an ideal solution, but it will convert the username to a valid name instead of giving you an error. i.e replace `_` with `-`.
